src/app/3d/qgspointcloudlayer3drendererwidget.cpp

```diff
@@ -0,0 +1,134 @@
+/***************************************************************************
+  qgspointcloudlayer3drendererwidget.cpp
+  ------------------------------
+  Date                 : November 2020
+  Copyright            : (C) 2020 by Nedjima Belgacem
+  Email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgspointcloudlayer3drendererwidget.h"
+
+#include "qgspointcloud3dsymbolwidget.h"
+#include "qgspointcloudlayer3drenderer.h"
+
+#include "qgspointcloudlayer.h"
+#include "qgspointcloud3dsymbol.h"
+
+#include <QBoxLayout>
+#include <QCheckBox>
+
+QgsPointCloudLayer3DRendererWidget::QgsPointCloudLayer3DRendererWidget( QgsPointCloudLayer *layer, QgsMapCanvas *canvas, QWidget *parent )
+  : QgsMapLayerConfigWidget( layer, canvas, parent )
+{
+  setPanelTitle( tr( "3D View" ) );
+
+  QVBoxLayout *layout = new QVBoxLayout( this );
+  layout->setContentsMargins( 0, 0, 0, 0 );
+
+  mChkEnabled = new QCheckBox( tr( "Enable 3D Renderer" ), this );
+  layout->addWidget( mChkEnabled );
+
+  mWidgetPointCloudSymbol = new QgsPointCloud3DSymbolWidget( layer, this );
+  layout->addWidget( mWidgetPointCloudSymbol );
+
+  connect( mChkEnabled, &QCheckBox::clicked, this, &QgsPointCloudLayer3DRendererWidget::onEnabledClicked );
+  connect( mWidgetPointCloudSymbol, &QgsPointCloud3DSymbolWidget::changed, this, &QgsPointCloudLayer3DRendererWidget::widgetChanged );
+}
+
+void QgsPointCloudLayer3DRendererWidget::setRenderer( const QgsPointCloudLayer3DRenderer *renderer )
+{
+  mRenderer.reset( renderer ? renderer->clone() : nullptr );
+  if ( renderer != nullptr )
+    mWidgetPointCloudSymbol->setSymbol( const_cast<QgsPointCloud3DSymbol *>( renderer->symbol() ) );
+  whileBlocking( mChkEnabled )->setChecked( renderer ? renderer->symbol()->isEnabled() : false );
+}
+
+QgsPointCloudLayer3DRenderer *QgsPointCloudLayer3DRendererWidget::renderer()
+{
+  // TODO: use unique_ptr for point cloud symbol
+  QgsPointCloud3DSymbol *sym = mWidgetPointCloudSymbol->symbol();
+  mRenderer.reset( new QgsPointCloudLayer3DRenderer );
+  mRenderer->setSymbol( sym );
+  delete sym;
+  mRenderer->setLayer( qobject_cast<QgsPointCloudLayer *>( mLayer ) );
+  return mRenderer.get();
```

Comment:
mRenderer doesn't seem to be used anywhere. I'd refine this and make renderer() return a new instance of the renderer (with the caller taking ownership). Then apply() can call `mLayer->setRenderer3D( renderer() )`